packages/jaeger-ui/src/components/TracePage/TraceSpanView/index.tsx

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { Component } from 'react';
+import { Row, Col, Table, Button, Select } from 'antd';
+import moment from 'moment';
+import { ColumnProps } from 'antd/es/table';
+import { SelectValue } from 'antd/lib/select';
+import FormItem from 'antd/lib/form/FormItem';
+import './index.css';
+import { TNil } from '../../../types';
+import { Trace, Span } from '../../../types/trace';
+
+const Option = Select.Option;
+
+function getNestedProperty(path: string, span: any): string {
+  return path.split('.').reduce((prev, curr) => {
+    return prev ? prev[curr] : null;
+  }, span);
+}
+
+function timeConversion(microseconds: number) {
+  const milliseconds: number = parseInt((microseconds / 1000).toFixed(2), 10);
+  const seconds: number = parseInt((milliseconds / 1000).toFixed(2), 10);
+  const minutes: number = parseInt((milliseconds / (1000 * 60)).toFixed(1), 10);
+  const hours: number = parseInt((milliseconds / (1000 * 60 * 60)).toFixed(1), 10);
+  const days: number = parseInt((milliseconds / (1000 * 60 * 60 * 24)).toFixed(1), 10);
+  let timeText;
+  if (microseconds < 1000) {
+    timeText = `${microseconds}μs`;
+  } else if (milliseconds < 1000) {
+    timeText = `${milliseconds}ms`;
+  } else if (seconds < 60) {
+    timeText = `${seconds}Sec`;
+  } else if (minutes < 60) {
+    timeText = `${minutes}Min`;
+  } else if (hours < 24) {
+    timeText = `${hours}Hrs`;
+  } else {
+    timeText = `${days}Days`;
+  }
+  return timeText;
+}
+
+type Props = {
+  trace: Trace;
+  uiFindVertexKeys: Set<string> | TNil;
+  uiFind: string | null | undefined;
+};
+
+type State = {
+  searchText: string;
+  searchedColumn: string;
+  data: Span[];
+  serviceNamesList: string[];
+  operationNamesList: string[];
+  serviceNameOperationsMap: Map<string, string[]>;
+  filtered: Record<string, string[]>;
+  selectedServiceName: string[];
+  selectedOperationName: string[];
+  filteredData: Span[];
+};
+
+export default class TraceSpanView extends Component<Props, State> {
+  constructor(props: Props, state: State) {
+    super(props, state);
+    const serviceNamesList = new Set<string>();
+    const operationNamesList = new Set<string>();
+    const serviceNameOperationsMap = new Map<string, string[]>();
+
+    this.props.trace.spans.map(span => {
```

Comment:
Why are you using a map here? I don't see where you use the mapped values.

Reply:
Removed this